src/circuit/listener.js

```diff
@@ -24,7 +29,7 @@ module.exports = (circuit) => {
   listener.listen = async (addr) => {
     const addrString = String(addr).split('/p2p-circuit').find(a => a !== '')
 
-    const relayConn = await circuit._dialer.connectToPeer(multiaddr(addrString))
+    const relayConn = await libp2p.dial(multiaddr(addrString))
```

Comment:
I changed this to reuse connections if already exist, instead of creating a new one. What do you think?